### PR TITLE
DOC: Add templates for issues and PRs

### PR DESCRIPTION
**Pull request by mattip**
Add a template for new issues or PRs. 

May be enhanced with the output of `numpy.show_config` if that gets us more info, see also issue #10983

A quick search on github and in the mailing list did not show open issues relating to this, did I miss a previous discussion?

The `ISSUE_TEMPLATE.md` is adapted from SciPy.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (eric-wieser) [inline .github/PULL_REQUEST_TEMPLATE.md]: Maybe better to put this in `<!-- -->` so that users don't submit this text?

Comment (mhvk) [reply .github/PULL_REQUEST_TEMPLATE.md]: By default, for a single commit, github will just use the commit message as initial text. This is probably worth mentioning, as it implies that with a good commit message you do not need a pull request template.

Comment (mattip) [reply .github/PULL_REQUEST_TEMPLATE.md]: I couldn't find a guide to the differences between text inside `<< >>` and `<!-- -->`, do you know why SciPy used `<< >>` ?

Comment (mattip) [reply .github/PULL_REQUEST_TEMPLATE.md]: So if it will not render, why would we want to use `<!-- -->`? It seems I don't understand the [review comment](https://github.com/numpy/numpy/pull/11512#discussion_r200475911) and how these templates are meant to work.

Comment (eric-wieser) [reply .github/PULL_REQUEST_TEMPLATE.md]: Because when someone creates a new issue, they see it pre-rendered. Anything that's an instruction to them, but is useless to us, should be visible only to them.

What I'm trying to avoid here is people submitting the template without modifications, and us seeing a lot of junk. Obviously they shouldn't do that, but I'd rather the result is a blank issue than one that clutters our search results.

Comment (mattip) [reply .github/PULL_REQUEST_TEMPLATE.md]: ok, fixing

**Comment by mattip** [comment]
Would be nice to see how this will work before merging

File: .github/ISSUE_TEMPLATE.md
Comment (eric-wieser) [inline .github/ISSUE_TEMPLATE.md]: I think providing
````
```python
import numpy as np
```
````

here would be helpful

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (eric-wieser) [inline .github/PULL_REQUEST_TEMPLATE.md]: Note that this markdown will never render - but I suppose the point of markdown is it's readable even without formatting,

File: .github/ISSUE_TEMPLATE.md
Comment (eric-wieser) [inline .github/ISSUE_TEMPLATE.md]: Something indicating that the entire template can be deleted for a feature request would be nice.

File: .github/ISSUE_TEMPLATE.md
Comment (mattip) [reply .github/ISSUE_TEMPLATE.md]: added

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (mattip) [reply .github/PULL_REQUEST_TEMPLATE.md]: removed markdown formatting

File: .github/ISSUE_TEMPLATE.md
Comment (mattip) [reply .github/ISSUE_TEMPLATE.md]: added

**Review by eric-wieser — APPROVED** [review]

**Comment by eric-wieser** [comment]
Not sure `CONTRIBUTING.md` needed moving, but I guess it doesn't matter much

**Comment by mattip** [comment]
This is a bit more than a DOC change so I am not going to merge it (as per policy that the PR originator should not merge unless it is a DOC change), but it seems ready to go.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (rgommers) [inline .github/PULL_REQUEST_TEMPLATE.md]: can you change "Be sure" to "Please be sure"

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (rgommers) [inline .github/PULL_REQUEST_TEMPLATE.md]: Can you change "Include a properly" to "We'd appreciate it if you would include a properly ..."

**Review by rgommers — CHANGES_REQUESTED** [review]
Not a huge fan of such templates, but they have some value - the tone needs to be friendly though.

**Review by rgommers — APPROVED** [review]

**Comment by rgommers** [comment]
LGTM now, thanks @mattip, @eric-wieser

**Comment by eric-wieser** [comment]
Github now has a [new mechanism](https://help.github.com/en/articles/about-issue-and-pull-request-templates) for handling this, that lets the user pick between multiple types of template.

We might want to add a template specifically for the `multiarray import failed` issue, among other things. Of note is that template issues can be auto-labelled.

**Comment by mattip** [comment]
We might want to have another template referring anaconda/intel issues to the proper issue tracker

**Pull request merged**
